front/src/app/shared/task-list-manager/task-list-manager.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { TaskGroupsList } from '../models/shared.models';
+import { ApiService } from '../api-service/api.service';
+import { BehaviorSubject, Observable, Subject, map, takeUntil } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskListManagerService {
+
+  listId$: Observable<string> = this.route.queryParams.pipe(
+    map((params): string => params['list']),
+  );
+
+  private allTaskListsSubject$ = new BehaviorSubject<TaskGroupsList | undefined>(undefined)

```

Comment:
BTW, as a solution to avoid having `undefined` in types and an actual `undefined` value, you can use `ReplaySubject(1)`